compiler/llvm/clangUtil.cpp

```diff
@@ -2832,8 +2832,10 @@ void runClang(const char* just_parse_filename) {
     // activate the GPU target
     splitStringWhitespace(generateClangGpuLangArgs(), clangOtherArgs);
 
-    std::string archFlag = std::string("--offload-arch=") + gpuArch;
-    clangOtherArgs.push_back(archFlag);
+    if (gpuArches.size() == 1) {
+      std::string archFlag = std::string("--offload-arch=") + *gpuArches.begin();
+      clangOtherArgs.push_back(archFlag);
+    }
```

Comment:
I've changed this to `>= 1`, and specifically changed `gpuArches` to be an ordered set. I hope that this will make assembly files with the least number of "recent" features, enabling both older and newer versions to compile fine. That is admittedly a bit of a hack.